src/host/inputBuffer.cpp

```diff
@@ -816,6 +816,12 @@ void InputBuffer::_HandleTerminalInputCallback(const TerminalInput::StringType&
 
         for (const auto& wch : text)
         {
+            if (wch == UNICODE_NULL)
+            {
+                // Simulates a null byte input event.
+                _storage.push_back(SynthesizeKeyEvent(true, 1, LOBYTE(OneCoreSafeVkKeyScanW(0)), 0, wch, CTRL_PRESSED | SHIFT_PRESSED));

```

Comment:
I don't think you can just hardcode the `CTRL_PRESSED` and `SHIFT_PRESSED` flags like this. You need to check what flags are set in the `HIBYTE` returned from the `VkKeyScanW(0)` call, and then convert those to equivalent `ControlKeyState` flags. Otherwise this will assumedly only work on keyboards where the null key is generated with a Ctrl Shift combination.

Reply:
In fact, no matter what keyboard it is, as long as a null byte is generated, it will be guaranteed to be converted back to a null byte in:
https://github.com/microsoft/terminal/blob/cc2ba5350d115b6898ac8b92482d0af7b8c4d365/src/host/stream.cpp#L234-L244

If we use the `HIBYTE` returned from the `VkKeyScanW(0)` as `dwControlKeyState`, it can't be converted back to a null byte. The value of `VkKeyScanW(0)` is `0x332`, same as `zeroKey`:
- The value of `CTRL_PRESSED` is `0x4 | 0x8`,  `WI_IsAnyFlagSet(0x332 & 0xFF00, 0x4 | 0x8) != WI_IsFlagSet(0x332, 0x200)`, and `WI_IsAnyFlagSet((0x332 & 0xFF00) >> 8, 0x4 | 0x8) != WI_IsFlagSet(0x332, 0x200)`.
- The value of `SHIFT_PRESSED` is `0x10`, `WI_IsAnyFlagSet(0x332 & 0xFF00, 0x10) != WI_IsFlagSet(zeroKey, 0x100)`, and `WI_IsAnyFlagSet((0x332 & 0xFF00) >> 8, 0x10) != WI_IsFlagSet(zeroKey, 0x100)`.

We have to pass `CTRL_PRESSED | SHIFT_PRESSED` as `dwControlKeyState`, to ensure that it will be converted back to a null byte. Otherwise we need to change `src/host/stream.cpp`, it may have side effects.

Reply:
I didn't mean you should pass the `HIBYTE` as the `dwControlKeyState` directly. You need to check the individual bits and convert them. If bit 0 is set, you include `SHIFT_PRESSED` in the `dwControlKeyState`; if bit 1 is set, you include `CTRL_PRESSED`, etc.

My concern is that not all keyboards will have `VkKeyScanW(0)` as `0x332`. For example, if it was `0x432`, and we've hardcoded the `dwControlKeyState` as `CTRL_PRESSED | SHIFT_PRESSED`, then that's assumedly not going to match.

Reply:
FWIW, when we use `VkKeyScanW()` or `OneCoreSafeVkKeyScanW()`, you follow [this](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-vkkeyscanw):

Return value | Description
-- | --
1 | Either SHIFT key is pressed.
2 | Either CTRL key is pressed.
4 | Either ALT key is pressed.
8 | The Hankaku key is pressed
16 | Reserved (defined by the keyboard layout driver).
32 | Reserved (defined by the keyboard layout driver).

These values correspond to the bit position where the modifier state is stored in `HIWORD(VkKeyScanW())`

So in our case, to read <kbd>ctrl</kbd> key state in `zeroKey`, you would do:

```cpp

bool ctrlState =  WI_IsFlagSet(HIWORD(zeroKey), 0x2);

// alternatively,
bool ctrlState =  WI_IsFlagSet(zeroKey, 0x200);
```

Reply:
@j4james @tusharsnx Thank you. 
I misunderstood. It has been fixed.